src/groupByKey.ts

```diff
@@ -2,6 +2,19 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey(
+  items: Array<string | number>,
+  key: string | number,
```

Comment:
The type of `key` should be `string` because it's used to access properties on an object, which are always strings or symbols. A `number` type for `key` is not appropriate here.